packages/neptune-extension/src/common/components/App.tsx

```diff
@@ -44,15 +52,22 @@ const App: React.FC<AppProps> = ({
           label="Upload"
           title="Upload to Neptune"
           icon="fa-cloud-upload"
-          visible={isApiTokenValid}
-          onClick={handleUpload}
+          visible={isApiTokenValid && notebookLoaded}
+          onClick={() => setUploadModalOpen(true)}
         />
       </ToolbarWrapper>
       {configureModalOpen && (
         <ConfigureModal
           onClose={() => setConfigureModalOpen(false)}
         />
       )}
+
+      { uploadModalOpen && (
+        <UploadForm
```

Comment:
There is some kind of name collision in here. I would expect if we see uploadModalOpen i would expect UploadModal instead of UploadForm. Please rename UploadForm to UploadModal

Reply:
I am ok with this convention, however `Form` seemed more general to me.

Reply:
Yes, but this compoenent extends ReactModal and not Form.

Reply:
To be precise, there is no extension here, because `extends` implies inheritance. This component is just a composite for the whole upload process, it consists of a form displayed in a modal. You change one line and it is displayed in something completely different. I would call it better `UploadProcedure`, but I have never seen such convention and would not like to introduce such. Neverheless I think `UploadModal` is *ok* for our purpose.